src/service/resources-impl.js

```diff
@@ -344,6 +344,7 @@ export class Resources {
     const resource = new Resource((++this.resourceIdCounter_), element, this);
     if (!element.id) {
       element.id = 'AMP_' + resource.getId();
+      element.setAttribute('amp-auto-id', element.id);
```

Comment:
Let's set it always. And let's call it `i-amp-id`. I would like `i-` to avoid anyone trying to use it for CSS or actions.